util/start_test

```diff
@@ -458,8 +460,9 @@ def run_sub_test(test=False):
 def generate_graphs(test=False):
     if test:
         basedir = os.path.dirname(test)
-        graph_files = [(test.replace(".chpl", "").replace(".test.c", "")
-                + ".graph")]
+        remove_dot = test.replace(".chpl", "").replace(".test.c", "")
+        remove_dot = remove_dot.replace(".ml-test.c", "")
+        graph_files = [(remove_dot + ".graph")]
```

Comment:
Maybe:

```python 
graph_files = [(os.path.splitext(test)[0] + ".graph")]
```

?

(not tested, but seems easier to split at any extension instead of hardcoding all of them here)

Reply:
We have tests in the test system with multiple extensions that aren't intended for special treatment by `start_test` that seem like they would cause problems for that strategy.  One example off the top of my head is `test/users/vass/type-tests.isSubtype.chpl`, but I think there are other ones

Reply:
splitext splits at the last `.`, so I think that should be fine.

Reply:
> splitext splits at the last .

Neat!  But then we'd run into the problem where we'd be looking for `foo.test.graph` instead of `foo.graph` for a performance test `foo.test.c`, right?

Reply:
I suppose I could call splitext twice for the `.test.c` and `.ml-test.c` files

Reply:
> Neat! But then we'd run into the problem where we'd be looking for foo.test.graph instead of foo.graph for a performance test foo.test.c, right?

Oh duh, yeah.. Ok scratch that.

Reply:
If you're up for it. Maybe start a `util/test/test_utils.py` file with:

```python
testable_exts = (".chpl", ".test.c", ".ml-test.c")

def has_testable_extension(f):
    return f.endswith(testable_exts)

def strip_testable_extension(f):
    for ext in testable_exts:
        if f.endswith(ext):
            return f[:-len(ext)]
    return f
```

and use that here, sub_clean, and sub_test.

If not. I'm fine with this as-is.